pkg/registry/core/service/rest.go

```diff
@@ -419,6 +422,9 @@ func (rs *REST) ResourceLocation(ctx genericapirequest.Context, id string) (*url
 	if err != nil {
 		return nil, nil, err
 	}
+	if !utilfeature.DefaultFeatureGate.Enabled(features.ServiceProxyAllowExternalIPs) {
```

Comment:
seems like we'd want to do the check on a particular address inside the loop, after we've verified the subset even has the port we want, since it involves a remote call

Reply:
Do you think it really matters? How many times is there an endpoint that doesn't match the port for the service?

(and is Proxy really a performance-sensitive call?)

Reply:
if you have a service backed by 400 pods (c.f.https://github.com/kubernetes/kubernetes/issues/58050 ), this would make it do 400 pod lookups as a pre-filter before selecting a single one and using it.

Reply:
That's not the point, because you're going to pay that cost no matter what. My point was: how many services do you not pay it either way?

Reply:
If you find a suitable endpoint address first, then verify the pod as a second step, in the normal case, you only look up one pod, right?

Reply:
Ok, I see your approach, I can restructure the code in that way, its more complicated then I was thinking...